File: app/src/main/java/ru/neverlands/abclient/db/AppDatabase.java
package ru.neverlands.abclient.db;

import android.content.Context;
import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;
import androidx.room.TypeConverters;
import ru.neverlands.abclient.model.Contact;
import ru.neverlands.abclient.model.MapTypeConverter;
import ru.neverlands.abclient.model.Thing;

@Database(entities = {Thing.class, Contact.class}, version = 4, exportSchema = false)
@TypeConverters(MapTypeConverter.class)
public abstract class AppDatabase extends RoomDatabase {

    public abstract ThingDao thingDao();
    public abstract ContactDao contactDao();

    private static volatile AppDatabase INSTANCE;

    public static AppDatabase getDatabase(final Context context) {
        if (INSTANCE == null) {
            synchronized (AppDatabase.class) {
                if (INSTANCE == null) {
                    INSTANCE = Room.databaseBuilder(context.getApplicationContext(),
                            AppDatabase.class, "abclient_database")
                            .fallbackToDestructiveMigration()
                            .build();
                }
            }
        }
        return INSTANCE;
    }
}

File: app/src/main/java/ru/neverlands/abclient/db/AppDatabase.kt
package ru.neverlands.abclient.db

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import androidx.room.TypeConverters
import ru.neverlands.abclient.model.MapTypeConverter
import ru.neverlands.abclient.model.Thing

@Database(entities = [Thing::class], version = 1, exportSchema = false)
@TypeConverters(MapTypeConverter::class)
abstract class AppDatabase : RoomDatabase() {

    abstract fun thingDao(): ThingDao

    companion object {
        @Volatile
        private var INSTANCE: AppDatabase? = null

        fun getDatabase(context: Context): AppDatabase {
            return INSTANCE ?: synchronized(this) {
                val instance = Room.databaseBuilder(
                    context.applicationContext,
                    AppDatabase::class.java,
                    "abclient_database"
                )
                .fallbackToDestructiveMigration()
                .build()
                INSTANCE = instance
                instance
            }
        }
    }
}
